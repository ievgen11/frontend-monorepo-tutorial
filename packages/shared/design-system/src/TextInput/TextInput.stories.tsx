import * as React from "react";
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import { TextInput } from "./index";

export default {
  title: "TextInput",
  component: TextInput,
  decorators: [withKnobs],
};

export const Default = () => (
  <TextInput
    disabled={boolean("Disabled", false)}
    onChange={action("onChange")}
    value={text("Tracking Number", "9400 1000 0000 0000 0000 00")}
    placeholder={text("Placeholder", "Enter Tracking Number")}
  />
);
