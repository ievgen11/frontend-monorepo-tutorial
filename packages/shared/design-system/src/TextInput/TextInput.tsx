import React, { useState } from "react";
import styled from "styled-components";

const Input = styled.input`
  width: 100%;
  font-family: inherit;
  background-color: #fff;
  border: none;
  color: #212529;
  padding: 12px 24px;
  font-weight: 500;
  font-size: 12px;
  white-space: nowrap;
  text-overflow: ellipsis;
  border-radius: 32px;
  box-sizing: border-box;
  box-shadow: -4px -4px 18px 6px #fff, 0 6px 12px 0 rgba(0, 0, 0, 0.2),
    6px 0 12px 0 rgba(0, 0, 0, 0.2), -6px 0 12px 0 rgba(0, 0, 0, 0.2);
  transition: all 0.1s ease-in-out;

  @media only screen and (min-width: 768px) {
    padding: 18px 32px;
    font-size: 14px;
  }

  &:focus,
  &:active {
    text-decoration: underline;
  }

  &:focus {
    outline-style: none;
  }

  &:disabled {
    cursor: not-allowed;
    background-color: #c8c8c8;
    color: #a0a0a0;
  }
`;

export interface TextInputProps {
  value?: string;
  onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  className?: string;
  disabled?: boolean;
  title?: string;
  placeholder?: string;
  type?: string;
}

export const TextInput: React.FC<TextInputProps> = ({
  value: initValue = "",
  onChange = (): null => null,
  disabled = false,
  placeholder,
  ...props
}): JSX.Element => {
  const [value, setValue] = useState(initValue);

  return (
    <Input
      type="text"
      value={value}
      disabled={disabled}
      onChange={(evt) => {
        setValue(evt.target.value);
        onChange(evt);
      }}
      placeholder={placeholder}
      {...props}
    />
  );
};
