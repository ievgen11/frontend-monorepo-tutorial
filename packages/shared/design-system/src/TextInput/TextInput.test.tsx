import * as React from "react";
import * as renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import { TextInput } from "./TextInput";

const TestID = "TextInputTestID";

describe("<TextInput />", () => {
  const testValues = {
    value: "1234-5678-90",
    onChange: jest.fn(),
  };

  test("Submit works", async () => {
    const { getByTestId } = render(
      <TextInput data-testid={TestID} {...testValues} />
    );

    const input = await getByTestId(TestID);

    fireEvent.change(input, { target: { value: testValues.value } });

    expect((input as HTMLInputElement).value).toBe(testValues.value);
  });

  test("Snapshot is intact", () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
