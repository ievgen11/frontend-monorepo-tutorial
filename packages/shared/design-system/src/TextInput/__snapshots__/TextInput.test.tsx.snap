// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Snapshot is intact 1`] = `
<input
  className="sc-bdfBwQ jWLat"
  disabled={false}
  onChange={[Function]}
  type="text"
  value=""
/>
`;
